value/src/main/java/com/google/auto/value/AutoValueExtension.java

```diff
@@ -0,0 +1,33 @@
+package com.google.auto.value;
+
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Created by rharter on 5/1/15.
+ */
+public interface AutoValueExtension {
+
+  public interface Context {
+    ProcessingEnvironment processingEnvironment();
+    String packageName();
+    TypeElement autoValueClass();
+    Map<String, ExecutableElement> properties();
+  }
+
+  public interface GeneratedClass {
+    String className();
+    String source();
+    Collection<String> additionalImports();
```

Comment:
This was implemented as strings to alleviate the need for extra dependencies.  For instance, in my Parcelable example, Parcelable and Parcel are both classes in the Android Framework.  By adding them as strings here, my plugin isn't dependent on the Android SDK, as these String FQCNs can simply be added to the imports clause.  AFAIK I can't create a TypeElement without having access to the represented class.